k2/csrc/fsa.h

```diff
@@ -77,13 +77,40 @@ struct Fsa {
   // contains the first arc-index leaving this state (index into `arcs`).
   // The next element of this array gives the end of that range.
   // Note: the final-state is numbered last, and implicitly has no
-  // arcs leaving it.
+  // arcs leaving it. For non-empty FSA, we put a duplicate of the final state
```

Comment:
On line 69, the comment is incorrect:
```
a final-state is represented by an arc with label == kEpsilon to final_state
```

It should be kFinalSymbol

Reply:
Oh, Yes, it's historical issue.